src/pkg/oci/common.go

```diff
@@ -63,6 +66,10 @@ func NewOrasRemote(url string) (*OrasRemote, error) {
 
 // WithRepository sets the repository for the remote as well as the auth client.
 func (o *OrasRemote) WithRepository(ref registry.Reference) error {
+	if o.root != nil {
```

Comment:
I feel like it is a weird pattern to have a non-reusable item on an otherwise reusable struct.

Reply:
Since a lot of operations call `FetchRoot`, I wanted to cache this request so that future calls do not need to perform an extraneous fetch. Could just always set `o.root = nil` here.